### PR TITLE
Added customizable metric port in helm chart

### PR DESCRIPTION
**Pull request by evalsocket**
Currently metric port is configurable but in helm there is no way to configure in helm, Added metric port in helm chart, Now user can configure the port. 

Tested All component with different port on sandbox, And it worked fine

Fix: https://github.com/flyteorg/flyte/issues/1864

### CODE REVIEW

File: charts/flyte-core/values.yaml [hunk 1]
```diff
@@ -200,6 +206,8 @@ flytepropeller:
   tolerations: []
   # -- affinity for Flytepropeller deployment
   affinity: {}
+  # Metric port used for collecting metric
+  metricPort: 10254
```
Comment (EngHabu) [inline charts/flyte-core/values.yaml:210]: You will have to put this in the config maps to take effect... it's not enough to list them here and in the container Ports...
e.g.: https://github.com/flyteorg/flytepropeller/blob/6a14e7fbffe89786fb1d8cde22715f93c2f3aff5/pkg/controller/config/config.go#L101

**Comment by evalsocket** [comment]
@pmahindrakar-oss you can review the PR

**Comment by pmahindrakar-oss** [comment]
@EngHabu  @evalsocket Do we have a prometheus deployment which can be configured to view these metrics. Also what usual scraping period of the metric port .

**Comment by evalsocket** [comment]
@pmahindrakar-oss We have datadog deployment, 

Scraping period will be configure on prometheus side.

**Comment by pmahindrakar-oss** [comment]
Ok . I wanted to see the scheduler metrics in datadog with this change.And also what's the scraping interval in datadog

**Review by EngHabu — DISMISSED** [review]

**Comment by kumare3** [comment]
We also have Prometheus deployed in the demo environment

File: charts/flyte/values.yaml [hunk 2]
```diff
@@ -357,6 +357,12 @@ flyte:
         - id: production
           name: production
 
+    # Refer to the full [structure](https://pkg.go.dev/github.com/lyft/flyteadmin@v0.3.37/pkg/runtime/interfaces#ApplicationConfig) for documentation.
+    schedulerConfig:
+      scheduler:
+        metricsScope: 'flyte:'
+        profilerPort: 10254
+
```
Comment (EngHabu) [inline charts/flyte/values.yaml:365]: We do not need this here, do we?
basically we do not need anything under `flyte:` in the flyte helm chart...

**Review by EngHabu — APPROVED** [review]

**Pull request merged**